apps/root/src/common/components/transaction-steps/index.tsx

```diff
@@ -243,77 +237,145 @@ const StyledTransactionStepButtonContainer = styled.div`
+
+    React.useEffect(() => {
+      if (hash && !isPendingTransaction && receipt && !hasConfirmedRef.current && onActionConfirmed) {
+        hasConfirmedRef.current = true;
+        onActionConfirmed(hash);
+      }
+    }, [hash, isPendingTransaction, receipt]);
+
+    const infiniteBtnText = (
+      <FormattedMessage
+        description="Allow us to use your coin (modal max)"
+        defaultMessage="Authorize Max {symbol}"
+        values={{
+          symbol: token.symbol,
+        }}
+      />
+    );
+
+    const specificBtnTextAsSecondary = (
+      <FormattedMessage
+        description="Allow us to use your coin (home exact secondary)"
+        defaultMessage="{amount} {symbol}"
+        values={{ symbol: token.symbol, amount: formatCurrencyAmount(amount, token, 4) }}
+      />
+    );
+
+    const specificBtnText = (
+      <FormattedMessage
+        description="Allow us to use your coin (home exact)"
+        defaultMessage="Authorize {amount} {symbol}"
+        values={{ symbol: token.symbol, amount: formatCurrencyAmount(amount, token, 4) }}
+      />
+    );
+
+    const waitingForAppvText = (
+      <FormattedMessage
+        description="waiting for approval"
+        defaultMessage="Waiting for your {symbol} to be authorized"
+        values={{
+          symbol: token.symbol,
+        }}
+      />
+    );
 
     return (
       <>
-        <StyledTransactionStepIcon isLast={isLast} isFirst={isFirst}>
-          <StyledTransactionStepIconContent>
-            <TokenIcon token={extraData.token} size="40px" />
+        <TransactionReceipt open={showReceipt} onClose={() => setShowReceipt(false)} transaction={receipt} />
+        <StyledTransactionStepIcon isLast={isLast} isCurrentStep={isCurrentStep}>
+          <StyledTransactionStepIconContent isCurrentStep={isCurrentStep}>
+            <WalletMoneyIcon />
           </StyledTransactionStepIconContent>
         </StyledTransactionStepIcon>
-        <StyledTransactionStepContent>
-          <Typography variant="body">
-            {extraData.isPermit2Enabled ? (
-              <FormattedMessage
-                description="transationStepApprove"
-                defaultMessage="{step} - Enable universal approval for {token}"
-                values={{ step, target: extraData.swapper, token: extraData.token.symbol }}
-              />
-            ) : (
-              <FormattedMessage
-                description="transationStepApprove"
-                defaultMessage="{step} - Allow {target} to use your {token}"
-                values={{ step, target: extraData.swapper, token: extraData.token.symbol }}
-              />
-            )}
-          </Typography>
-          <Typography variant="bodySmall">
-            <Address trimAddress address={account || ''} />
-          </Typography>
-          {isCurrentStep && (
-            <StyledTransactionStepButtonContainer>
-              <AllowanceSplitButton
```

Comment:
Why did we remove this component>?

Reply:
I find both allowance split button + allowance tooltip useless in the new design. Should we keep them?